### PR TITLE
Implements the experimental contains filter operator

### PR DESCRIPTION
**Pull request by irevoire**
# Pull Request
Related PRD: (private link) https://www.notion.so/meilisearch/Contains-Like-Filter-Operator-0d8ad53c6761466f913432eb1d843f1e
Public usage page: https://meilisearch.notion.site/Contains-filter-operator-usage-3e7421b0aacf45f48ab09abe259a1de6

## Related issue
Fixes https://github.com/meilisearch/meilisearch/issues/3613

## What does this PR do?
- Extract the contains operator from this PR: https://github.com/meilisearch/meilisearch/pull/3751
- Gate it behind a feature flag
- Add tests

### CODE REVIEW

File: meilisearch/src/routes/multi_search.rs [hunk 1]
```diff
@@ -112,6 +112,7 @@ pub async fn multi_search_with_post(
                         ));
                     }
 
+                    let features = index_scheduler.features();
```
Comment (dureuill) [inline meilisearch/src/routes/multi_search.rs:115]: I don't think we should be getting the features another time here (they could have changed!).

Rather, we should reuse the top-level `features` binding.

File: meilisearch/src/search/federated.rs [hunk 2]
```diff
@@ -473,8 +473,14 @@ pub fn perform_federated_search(
...
 
-                let (mut search, _is_finite_pagination, _max_total_hits, _offset) =
-                    prepare_search(&index, &rtxn, &query, &search_kind, time_budget)?;
+                let (mut search, _is_finite_pagination, _max_total_hits, _offset) = prepare_search(
+                    &index,
+                    &rtxn,
+                    &query,
+                    &search_kind,
+                    time_budget,
+                    index_scheduler.features(),
```
Comment (dureuill) [inline meilisearch/src/search/federated.rs:482]: we should reuse the features that were passed as parameters here

File: meilisearch/src/routes/multi_search.rs (hunk 1 above)
Comment (irevoire) [reply meilisearch/src/routes/multi_search.rs:115]: Oops yes I missed it

**Review by dureuill — APPROVED** [review]
:+1:

**Comment by irevoire** [comment]
bors merge

**Pull request merged**